packages/astro/src/runtime/server/index.ts

```diff
@@ -182,6 +182,13 @@ function formatList(values: string[]): string {
 
 const rendererAliases = new Map([['solid', 'solid-js']]);
 
+/** @internal */
+export function __astro_tag_component__(Component: unknown, rendererName: string) {
```

Comment:
What is this being used for?

Reply:
Is this being used to allow an Astro component to be imported into MDX?

Reply:
Added comments in [38bec11](https://github.com/withastro/astro/pull/3859/commits/38bec1142125fa3109d7e9767d54633e53224d0c)

Reply:
We used to do this inside of rendering by storing in a WeakMap. I can't remember why this got removed (I would assume it wasn't on purpose). I feel like that old way is more resilient as it doesn't rely on AST parsing and works regard of whether the component is JSX or anything else.